relay-server/src/envelope.rs

```diff
@@ -1039,6 +1039,13 @@ pub struct EnvelopeHeaders<M = RequestMeta> {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     trace: Option<ErrorBoundary<DynamicSamplingContext>>,
 
+    /// A list of features required to process this envelope.
+    ///
+    /// This is an internal field that should only be set by Relay.
+    /// It is a serializable header such that it persists when spooling.
+    #[serde(default, skip_serializing_if = "SmallVec::is_empty")]
```

Comment:
Should we serialize and deserialize this at all?

```suggestion
    #[serde(default, skip)]
```

Reply:
I wanted to make it `skip` initially, but this header is checked after disk spooling (see doc comment). This might actually be a problem for other `skip` (item) headers as well.

Reply:
Oops, didn't read the docs. 👍